extensions/wikia/Rail/RailController.class.php

```diff
@@ -8,139 +8,164 @@ class RailController extends WikiaController {
+			'wgAllowUserJs' => false,
+			'wgUseSiteJs' => false,
 			// override original wgTitle from title given in parameters
 			// we cannot use wgTitle that is created on by API because it's broken on wikis without '/wiki' in URL
 			// https://wikia-inc.atlassian.net/browse/BAC-906
-			$oldWgTitle = $wgTitle;
-			$wgTitle = $title;
-			$assetManager = AssetsManager::getInstance();
-			$railModules = $this->filterModules((new BodyController)->getRailModuleList(), self::FILTER_LAZY_MODULES);
-			$this->railLazyContent = '';
-			krsort($railModules);
-			foreach ($railModules as $railModule) {
-				$this->railLazyContent .= $this->app->renderView(
-					$railModule[0], /* Controller */
-					$railModule[1], /* Method */
-					$railModule[2] /* array of params */
-				);
-			}
+			'wgTitle' => $title
+		] );
 
-			$this->railLazyContent .= Html::element('div', ['id' => 'WikiaAdInContentPlaceHolder']);
+		$assetManager = AssetsManager::getInstance();
 
-			$this->css = $sassFiles = [];
-			foreach (array_keys($this->app->wg->Out->styles) as $style) {
-				if ($wgAllInOne && $assetManager->isSassUrl($style)) {
-					$sassFiles[] = $style;
-				} else {
-					$this->css[] = $style;
-				}
-			}
+		$railModuleListRaw = $wrapper->wrap( function () {
+			return ( new BodyController )->getRailModuleList();
+		} );
 
-			if (!empty($sassFiles)) {
-				$excludeScss = (array) $this->getRequest()->getVal('excludeScss', []);
-				$sassFilePath = (array) $assetManager->getSassFilePath($sassFiles);
-				$includeScss = array_diff($sassFilePath, $excludeScss);
+		$railModules = $this->filterModules(
+			$railModuleListRaw,
+			self::FILTER_LAZY_MODULES
+		);
+		$this->railLazyContent = '';
 
-				// SUS-771: Log any duplicate CSS that rail modules try to load but are already loaded by Oasis skin
-				$duplicateScss = array_intersect( $sassFilePath, $excludeScss );
-				if ( count( $duplicateScss ) ) {
-					Wikia\Logger\WikiaLogger::instance()->info( 'SUS-771', [
-						'styles' => json_encode( $duplicateScss )
-					] );
-				}
+		krsort( $railModules );
 
-				if (!empty($includeScss)) {
-					$this->css[] = $assetManager->getSassesUrl($includeScss);
-				}
-			}
+		// TODO XW-2760 remove after experiment is done
+		$isAdMixExperimentEnabled = $this->request->getVal( 'isAdMixExperimentEnabled', false );
 
-			// Do not load user and site jses as they are already loaded and can break page
-			$oldWgUseSiteJs = $wgUseSiteJs;
-			$oldWgAllowUserJs = $wgAllowUserJs;
-			$wgUseSiteJs = false;
-			$wgAllowUserJs = false;
+		if ( $isAdMixExperimentEnabled ) {
+			// We should display only TOP_RIGHT_BOXAD & INCONTENT_BOXAD_1 in the experiment's right rail
+			// Some user scripts depend on NATIVE_TABOOLA_RAIL to be there, though (e.g. Discord)
+			// We remove LEFT_SKYSCRAPER_2 because it's often displayed and disturbs the testing
+			// FIXME for AdEng - decide if this is fine, if yes then remove this comment line
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this "FIXME" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)